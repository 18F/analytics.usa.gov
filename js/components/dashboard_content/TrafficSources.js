import React from "react";
import PropTypes from "prop-types";

import ConsolidatedBarChart from "../chart/ConsolidatedBarChart";
import Tooltip from "../tooltip/Tooltip";

/**
 * Contains charts and other data visualizations for the traffic sources
 * section of the site. This component is mainly laying out the structure for
 * the section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function TrafficSources({ dataHrefBase }) {
  return (
    <div className="padding-top-1 grid-row">
      <div
        id="session_channel_groups"
        className="desktop:grid-col-6 desktop:padding-right-2"
      >
        <div className="chart__title">
          <a href="/definitions#dimension_default_channel_group">
            <Tooltip
              position="top"
              content="Channel groupings are rule-based definitions of your traffic sources."
            >
              Top Channels
            </Tooltip>
          </a>
          <a
            href={`${dataHrefBase}/top-session-channel-group-30-days.csv`}
            aria-label="top-session-channel-group-30-days.csv"
          >
            <svg
              className="usa-icon margin-bottom-neg-05 margin-left-05"
              aria-hidden="true"
              focusable="false"
              role="img"
            >
              <use xlinkHref="/assets/uswds/img/sprite.svg#file_present"></use>
            </svg>
          </a>
        </div>
        <div className="text--capitalize">
          <ConsolidatedBarChart
            dataUrl={`${dataHrefBase}/top-session-channel-group-30-days.json`}
            chartDataKey={"session_default_channel_group"}
            maxItems={10}
          />
        </div>
      </div>

      <div
        id="session_source_mediums"
        className="desktop:grid-col-6 desktop:padding-left-2"
      >
        <div className="chart__title">
          <a href="/definitions#dimension_source_medium">
            <Tooltip
              position="top"
              content="A source is a representation of the publisher or inventory source from which traffic originated. A medium is a method for acquiring users to websites or applications."
            >
              Top Sources/Media
            </Tooltip>
          </a>
          <a
            href={`${dataHrefBase}/top-session-source-medium-30-days.csv`}
            aria-label="top-session-source-medium-30-days.csv"
          >
            <svg
              className="usa-icon margin-bottom-neg-05 margin-left-05"
              aria-hidden="true"
              focusable="false"
              role="img"
            >
              <use xlinkHref="/assets/uswds/img/sprite.svg#file_present"></use>
            </svg>
          </a>
        </div>
        <div className="text--capitalize">
          <ConsolidatedBarChart
            dataUrl={`${dataHrefBase}/top-session-source-medium-30-days.json`}
            chartDataKey={"session_source_medium"}
            maxItems={10}
          />
        </div>
      </div>
    </div>
  );
}

TrafficSources.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default TrafficSources;
