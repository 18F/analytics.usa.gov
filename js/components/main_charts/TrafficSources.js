import React from "react";
import PropTypes from "prop-types";

import TopChannels from "./TopChannels";
import TopSourceMedia from "./TopSourceMedia";

/**
 * Contains charts and other data visualizations for the traffic sources
 * section of the site. This component is mainly laying out the structure for
 * the section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function TrafficSources({ dataHrefBase }) {
  return (
    <>
      <section
        id="session_channel_groups"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Channels</h4>
        <TopChannels dataHrefBase={dataHrefBase} />
      </section>

      <section
        id="session_source_mediums"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Sources/Media</h4>
        <TopSourceMedia dataHrefBase={dataHrefBase} />
      </section>
    </>
  );
}

TrafficSources.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default TrafficSources;
