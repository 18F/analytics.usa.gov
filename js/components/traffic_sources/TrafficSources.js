import React from "react";
import PropTypes from "prop-types";

import TopChannels from "./TopChannels";
import TopSourceMedia from "./TopSourceMedia";

function TrafficSources({ dataURL, dataPrefix }) {
  const dataHrefBase = `${dataURL}/${dataPrefix}`;

  return (
    <>
      <section
        id="session_channel_groups"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Channels</h4>
        <TopChannels dataHrefBase={dataHrefBase} />
      </section>

      <section
        id="session_source_mediums"
        className="desktop:grid-col-6 bar-chart-component"
      >
        <h4>Top Sources/Media</h4>
        <TopSourceMedia dataHrefBase={dataHrefBase} />
      </section>
    </>
  );
}

TrafficSources.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
};

export default TrafficSources;
