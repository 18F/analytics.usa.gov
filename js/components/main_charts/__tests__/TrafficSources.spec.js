import { create, act } from "react-test-renderer";
import TrafficSources from "../TrafficSources";

describe("TrafficSources", () => {
  it("renders", () => {
    let component;
    act(() => {
      component = create(
        <TrafficSources dataHrefBase="http://www.example.com/data/" />,
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
