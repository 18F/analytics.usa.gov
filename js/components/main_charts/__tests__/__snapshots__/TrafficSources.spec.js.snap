// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrafficSources renders 1`] = `
<DocumentFragment>
  <section
    class="desktop:grid-col-6 bar-chart-component"
    id="session_channel_groups"
  >
    <h4>
      Top Channels
    </h4>
    <figure
      id="chart_session_channel_group"
    >
      <div
        class="data bar-chart"
      />
    </figure>
  </section>
  <section
    class="desktop:grid-col-6 bar-chart-component"
    id="session_source_mediums"
  >
    <h4>
      Top Sources/Media
    </h4>
    <figure
      id="chart_session_source_medium"
    >
      <div
        class="data bar-chart"
      />
    </figure>
  </section>
</DocumentFragment>
`;
